src/core/config/config.rs

```diff
@@ -210,53 +210,11 @@ pub struct RootSchema {
 /// Used to omit a field from public consumption.
 pub struct Omit {}
 
-// generate Resolver with macro in order to autogenerate conversion code
-// from the underlying directives.
-// TODO: replace with derive macro
-macro_rules! create_resolver {
-    ($($var:ident($ty:ty)),+$(,)?) => {
-        #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, schemars::JsonSchema)]
-        #[serde(rename_all = "camelCase")]
-        pub enum Resolver {
-            // just specify the same variants
-            $($var($ty),)+
-        }
-
-        impl Resolver {
-            pub fn from_directives(
-                directives: &[Positioned<ConstDirective>],
-            ) -> Valid<Option<Self>, String> {
-                let mut result = None;
-                let mut resolvable_directives = Vec::new();
-                let mut valid = Valid::succeed(());
-
-                $(
-                    // try to parse directive from the Resolver variant
-                    valid = valid.and(<$ty>::from_directives(directives.iter()).map(|resolver| {
-                        if let Some(resolver) = resolver {
-                            // on success store it as a result and remember parsed directives
-                            result = Some(Self::$var(resolver));
-                            resolvable_directives.push(<$ty>::trace_name());
-                        }
-                    }));
-                )+
-
-                valid.and_then(|_| {
-                    if resolvable_directives.len() > 1 {
-                        Valid::fail(format!(
-                            "Multiple resolvers detected [{}]",
-                            resolvable_directives.join(", ")
-                        ))
-                    } else {
-                        Valid::succeed(result)
-                    }
-                })
-            }
-        }
-    };
-}
-
-create_resolver! {
+#[derive(
+    Serialize, Deserialize, Clone, Debug, PartialEq, Eq, schemars::JsonSchema, CustomResolver,
```

Comment:
We are essentially deriving `DirectiveCodec`, can we use that?

Reply:
we can't do it right now because we need a custom implementation of `DirectiveCodec` here and the default one `impl<'a, A: Deserialize<'a> + Serialize + 'a> DirectiveCodec for A` doesn't allow to write specialized implementation